### PR TITLE
Fixed bug with validation sets having additional classes.

### PR DESCRIPTION
**Pull request by hoytak**
This PR adds tests for validation classes having additional classes not
seen in the training classes.  These tests exposed problems with both
the evaluation and exporting to CoreML, both of which are fixed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -213,7 +213,7 @@ message(WARNING "OpenMP Libraries were not found")
 #*                                                                        */
 #**************************************************************************/
 if (CLANG)
-set(CPP11_FLAGS "-std=c++11 -stdlib=libc++ -Wno-deprecated-register -Wno-enum-compare -Wno-conversion-null -Wno-constant-logical-operand -Wno-parentheses-equality -ftemplate-depth=900" CACHE STRING "C++11 enabling flags")
+  set(CPP11_FLAGS "-std=c++11 -stdlib=libc++ -Wno-stdlibcxx-not-found -Wno-deprecated-register -Wno-enum-compare -Wno-conversion-null -Wno-constant-logical-operand -Wno-parentheses-equality -ftemplate-depth=900" CACHE STRING "C++11 enabling flags")
```
Comment (znation) [inline CMakeLists.txt]: Why is this needed?

Comment (hoytak) [reply CMakeLists.txt]: The version of clang shipping with the latest Mohave beta warns on if we use libc++ instead of libcxx, which causes compilation to fail with -Werror.

Comment (znation) [reply CMakeLists.txt]: What do you mean, `libc++` instead of `libcxx`? Are those two different names for the same thing, and one is deprecated? If that's the case, can we just use `libcxx` all the time instead of not warning in this situation?
